### PR TITLE
Added home dir ~/.flyte in search path

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Currently the config.yaml file is searched in the following locations.

- currentDir
- /etc/flyte/config
- $GOPATH/src/github.com/lyft/flytestdlib

To make it compatible with flytekit we should also add homedir/.flyte directory to search path.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/941

## Follow-up issue
_NA_

### CODE REVIEW

**Review by katrogan — APPROVED** [review]

File: config/files/finder.go [hunk 1]
```diff
@@ -13,8 +13,16 @@ const (
 var configLocations = [][]string{
 	{"."},
 	{"/etc", "flyte", "config"},
+	{UserHomeDir(), ".flyte"},
 	{os.ExpandEnv("$GOPATH"), "src", "github.com", "lyft", "flytestdlib"},
 }
+var osUserHomDir = os.UserHomeDir
```
Comment (kumare3) [inline config/files/finder.go:19]: why a variable? Why not directly use it in the function?

Comment (kumare3) [reply config/files/finder.go:19]: I dont know if we should add the home dir directly here. I would prefer that to be done only for flytectl

Comment (kumare3) [reply config/files/finder.go:19]: can we not just pass the homeDir location to `FindConfigFiles`.
The reason is - this config loading is general purpose and used across all of Flyte.

Comment (pmahindrakar-oss) [reply config/files/finder.go:19]: Variable to help with  unit testing  so that i dont have to rely to the filesystem call.
FakeUserHomeDir function mocks the behavior.

Right now its used by flytectl only so yeah will move it to that repo only.
